api/src/main/java/org/openmrs/module/attachments/obs/DefaultAttachmentHandler.java

```diff
@@ -39,20 +36,13 @@ protected ComplexData readComplexData(Obs obs, ValueComplex valueComplex, String
 				complexData.getData(), valueComplex.getMimeType()).asComplexData();
 	}
 
-	protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData) {
-		// We use a temp obs whose value complex points to the file name
-		Obs tmpObs = new Obs();
-		tmpObs.setValueComplex(complexData.asComplexData().getTitle()); // Temp obs used as a safety
-		return getParent().purgeComplexData(tmpObs);
+	protected boolean deleteComplexData(Obs obs) {
+		return getParent().purgeComplexData(obs);
```

Comment:
I'm not *why* we are relying on instantiating a temp obs to delete, but it didn't seem to be needed.